src/components/color-picker/color-picker.tsx

```diff
@@ -293,7 +293,7 @@ export class ColorPicker implements InteractiveComponent {
 
   private hueScopeNode: HTMLDivElement;
 
-  private internalColorUpdateContext: "internal" | "initial" | null = null;
+  private internalColorUpdateContext: "internal" | "initial" | "user-interaction" | null = null;
```

Comment:
> a good 'ol interface would be good to use to specify value updating context,

100%! I'll create a refactor/research issue today for this.

> The hard part is figuring out what to name these different contexts... 🤔

Yeah. The common contexts I've observed are: load-time, internal and user-interaction. Do you have additional examples?

Reply:
> Yeah. The common contexts I've observed are: load-time, internal and user-interaction. Do you have additional examples?

I've differentiated between initial loading/mounting stage, internal (which implies user interaction), and external and in some special contexts where there is an internal component like `input-time-picker/time-picker` I identified value changes coming from `time-picker`.  Adding an additional "user interaction" context is interesting, but it would be good to iron out the difference between "internal" and "user interaction" if there is any.

Reply:
Here is the time picker example: https://github.com/Esri/calcite-components/blob/a6d110888f026f9df4fcadd8b0fc10225fe22d14/src/components/input-time-picker/input-time-picker.tsx#L297